web/packages/teleterm/electron-builder-config.js

```diff
@@ -68,6 +68,13 @@ module.exports = {
   },
   files: [
     'build/app/dist',
+    // main and renderer are put in their own folders in build/app/dist in the packaged app, so
+    // let's put agentCleanupDaemon in its own folder as well.
```

Comment:
I considered agentCleanupDaemon to be a separate process and I think I wanted to reflect the structure of `teleterm/src` in `build/app/dist`. But I totally forgot that shared process is already put into `build/app/dist/main`.

I haven't thought about creating a separate entry point for Webpack because in my mind this was equal to using TypeScript for the agent cleanup daemon. But after reading your comment I realized this is not necessarily true. Still, it'd be best for Webpack to not apply any transformations to that file so that we know that what we run in tests is equal to what will be run in prod.

I'll try to put it as a separate entrypoint. This should take care of the issues between prod and dev builds you mentioned.

---

The folder structure you proposed seems fine. I'd probably make it even more granular by mapping the top folders to processes, so the shared process and the agent cleanup daemon would get their own folders as well. Preload would technically be considered to belong to the renderer.